### PR TITLE
[CPU] Added H-Swish activation

### PR DESCRIPTION
**Pull request by a-sidorova**
Issue 34439

Added H-Swish activation:

- merge Power [x2], Clamp and Multiply to H-Swish
- supported fusing of H-Swish activation

PR in OneDNN: https://github.com/openvinotoolkit/oneDNN/pull/10

### CODE REVIEW

File: inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp [hunk 1]
```diff
@@ -658,6 +662,134 @@ void MKLDNNGraphOptimizer::MergeSigmoidAndMultiplyToSwish(MKLDNNGraph& graph) {
...
+        auto *powerAddNodePtr = dynamic_cast<MKLDNNPowerNode *>(powerAddNode.get());
+        if (powerAddNodePtr == nullptr)
+            THROW_IE_EXCEPTION << "Cannot cast " << powerAddNode->getName() << " to Power node";
+        auto *powerAddLayer = dynamic_cast<PowerLayer*>(powerAddNode->getCnnLayer().get());
+        if (powerAddLayer == nullptr)
+            THROW_IE_EXCEPTION << "Cannot get power layer " << powerAddNode->getName();
+        if (powerAddLayer->power != 1.0f || powerAddLayer->scale != 1.0f || powerAddLayer->offset != 3.0f)
+            return false;
+
+        auto *activationNodePtr = dynamic_cast<MKLDNNActivationNode *>(activationNode.get());
```
Comment (antonvor) [inline inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp:713]: I did not see the parameter check for the Clamp. Could you please add it?)

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/hswish.cpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            InferenceEngine::Precision::FP32
+    };
+
+    INSTANTIATE_TEST_CASE_P(Check, HSwishTest,
+                            ::testing::Combine(
+                                    ::testing::ValuesIn(netPrecisions),
+                                    ::testing::Values(std::vector<size_t>({1, 5, 8, 8})),
+                                    ::testing::Values(CommonTestUtils::DEVICE_CPU)),
+                            HSwishTest::getTestCaseName);
+}  // namespace
```
Comment (antonvor) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/hswish.cpp:23]: Add empty line in to end of file

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/hswish.cpp [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+#include <vector>
+
+#include "subgraph_tests/hswish.hpp"
+#include "common_test_utils/test_constants.hpp"
+
+using namespace LayerTestsDefinitions;
+
+namespace {
+    const std::vector<InferenceEngine::Precision> netPrecisions = {
```
Comment (antonvor) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/hswish.cpp:13]: remove indents in namespace section

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/hswish.cpp [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            InferenceEngine::Precision::FP32
+    };
+
+    INSTANTIATE_TEST_CASE_P(Check, HSwishTest,
+                            ::testing::Combine(
```
Comment (antonvor) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/hswish.cpp:18]: You are testing one case, can you rewrite this without  "Combine" please?

File: inference-engine/tests/functional/plugin/shared/include/subgraph_tests/hswish.hpp [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+namespace LayerTestsDefinitions {
+
+class HSwishTest : public testing::WithParamInterface<shapeOfParams>,
+          virtual public LayerTestsUtils::LayerTestsCommon {
+public:
+    static std::string getTestCaseName(const testing::TestParamInfo<shapeOfParams> &obj);
+protected:
+    void SetUp() override;
+};
+} // namespace LayerTestsDefinitions
```
Comment (antonvor) [inline inference-engine/tests/functional/plugin/shared/include/subgraph_tests/hswish.hpp:26]: empty line

File: inference-engine/tests/functional/plugin/shared/src/subgraph_tests/hswish.cpp [hunk 6]
```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "subgraph_tests/hswish.hpp"
+
+namespace LayerTestsDefinitions {
+
```
Comment (antonvor) [inline inference-engine/tests/functional/plugin/shared/src/subgraph_tests/hswish.cpp:8]: indents

**Comment by antonvor** [comment]
And squash commits please

**Comment by a-sidorova** [comment]
@antonvor, thank you for your comments. I fixed all defects

**Comment by antonvor** [comment]
@dmitry-gorokhov could you finally review please?

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/hswish.cpp [hunk 7]
```diff
@@ -0,0 +1,22 @@
+// Copyright (C) 2019 Intel Corporation
```
Comment (iefode) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/hswish.cpp:1]: Why `hswish` single layer test did not created/instantiated?

Comment (a-sidorova) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/hswish.cpp:1]: HSwish is not yet in the opset. So this test covers only plugin optimization

**Comment by lazarevevgeny** [comment]
There is a [PR](https://github.com/openvinotoolkit/openvino/pull/1770) where HSwish is enabled in nGraph. Also, it contains fusing patterns to create HSwish from various sub-graphs so no need to implement these fusings in the CPU plugin. When the PR is merged the HSwish will be decomposed into primitives because the operation is not yet enabled in CPU plugin. Please, add one line like here https://github.com/openvinotoolkit/openvino/pull/1799/files#diff-14f7a6897b4719c8269edf3297c2a94dR84 to disable decomposition of HSwish for CPU plugin.

**Comment by lazarevevgeny** [comment]
@a-sidorova, @antonvor the PR with HSwish enabling https://github.com/openvinotoolkit/openvino/pull/1770 has been merged. Please, make changes like here https://github.com/openvinotoolkit/openvino/pull/1799/files#diff-14f7a6897b4719c8269edf3297c2a94dR84 in this PR to not decompose HSwish for the CPU plugin.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]
Please, make changes like this https://github.com/openvinotoolkit/openvino/pull/1799/files#diff-14f7a6897b4719c8269edf3297c2a94dR84 to not decompose HSwish in the CPU plugin.

File: inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp [hunk 8]
```diff
@@ -17,6 +17,7 @@
 #include "nodes/mkldnn_quantize_node.h"
 #include "nodes/mkldnn_mvn_node.h"
 #include "nodes/mkldnn_resample_node.h"
+#include "nodes/mkldnn_power_node.h"
```
Comment (antonvor) [inline inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp:20]: why `#include "nodes/mkldnn_power_node.h"`?

File: inference-engine/src/transformations/include/transformations/hswish_fusion.hpp [hunk 9]
```diff
@@ -61,3 +63,12 @@ class ngraph::pass::HSwishFusion: public ngraph::pass::GraphRewrite {
 public:
     HSwishFusionWithoutRelu();
 };
+
+/**
+ * @ingroup ie_transformation_common_api
+ * @brief HSwishFusion transformation replaces a sub-graph x * ((Clamp(x + 3), 0, 6) * const(1/6)) with a HSwish op.
```
Comment (antonvor) [inline inference-engine/src/transformations/include/transformations/hswish_fusion.hpp:69]: Maybe `Clamp(x + 3, 0, 6)`?

File: inference-engine/src/transformations/src/transformations/hswish_fusion.cpp [hunk 10]
```diff
@@ -178,3 +178,46 @@ ngraph::pass::HSwishFusionWithoutRelu::HSwishFusionWithoutRelu() {
     auto m = std::make_shared<ngraph::pattern::Matcher>(mul, "HSwishWithoutReluFusion");
     register_matcher(m, callback);
 }
+
+ngraph::pass::HSwishFusionWithClamp::HSwishFusionWithClamp() {
+    // Replaces a sub-graph x * ((Clamp(x + 3), 0, 6) * const(1/6)) with a HSwish op.
```
Comment (antonvor) [inline inference-engine/src/transformations/src/transformations/hswish_fusion.cpp:183]: clamp

Comment (a-sidorova) [reply inference-engine/src/transformations/src/transformations/hswish_fusion.cpp]: Thank you for review. Done

**Comment by a-sidorova** [comment]
> Please, make changes like this https://github.com/openvinotoolkit/openvino/pull/1799/files#diff-14f7a6897b4719c8269edf3297c2a94dR84 to not decompose HSwish in the CPU plugin.

@lazarevevgeny, thank you. Done

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/hswish.cpp (hunk 7 above)
Comment (a-sidorova) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/hswish.cpp]: HSwish has been added in the opset. So there is now a single layer test for hswish

**Review by dmitry-gorokhov — APPROVED** [review]

**Comment by dmitry-gorokhov** [comment]
CPU part looks good for me.

@lazarevevgeny could you please review NG transformation changes?

**Comment by lazarevevgeny** [comment]
> CPU part looks good for me.
> 
> @lazarevevgeny could you please review NG transformation changes?

@GlebKazantaev is a better person to review the nGraph transformation :)

**Review by lazarevevgeny — APPROVED** [review]
Overall changes looks good. But I would like to see the approve from @GlebKazantaev before the merge.

File: inference-engine/tests/ngraph_functions/src/activation.cpp [hunk 11]
```diff
@@ -93,6 +93,8 @@ std::shared_ptr<ngraph::Node> makeActivation(const ngraph::Output<Node> &in,
         }
         case ngraph::helpers::ActivationTypes::Mish:
             return std::make_shared<ngraph::op::v4::Mish>(in);
+        case ngraph::helpers::ActivationTypes::HSwish:
+            return std::make_shared<ngraph::op::v4::HSwish>(in);
```
Comment (iefode) [inline inference-engine/tests/ngraph_functions/src/activation.cpp:97]: Please rebase your branch on latest master. I merged https://github.com/openvinotoolkit/openvino/pull/1861/files to the master branch recently

**Review by iefode — APPROVED** [review]
Test part LGTM

File: inference-engine/tests/ngraph_functions/src/activation.cpp (hunk 11 above)
Comment (a-sidorova) [reply inference-engine/tests/ngraph_functions/src/activation.cpp]: Thank you, I rebased this branch on master

**Pull request merged**